scipy-stubs/signal/_peak_finding.pyi

```diff
@@ -1,37 +1,72 @@
 
-def argrelmin(data: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def argrelmax(data: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def argrelextrema(data: Untyped, comparator: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def peak_prominences(x: Untyped, peaks: Untyped, wlen: Untyped | None = None) -> Untyped: ...
+_Mode: TypeAlias = Literal["clip", "wrap"]
+_ProminencesResult: TypeAlias = tuple[npt.NDArray[np.float64], npt.NDArray[np.intp], npt.NDArray[np.intp]]
+_WidthsResult: TypeAlias = tuple[
+    npt.NDArray[np.float64], npt.NDArray[np.float64], npt.NDArray[np.float64], npt.NDArray[np.float64]
+]
+
+class _FindPeaksResultsDict(TypedDict):
+    peak_heights: NotRequired[npt.NDArray[np.float64]]
+    left_thresholds: NotRequired[npt.NDArray[np.float64]]
+    right_thresholds: NotRequired[npt.NDArray[np.float64]]
+    prominences: NotRequired[npt.NDArray[np.float64]]
+    left_bases: NotRequired[npt.NDArray[np.intp]]
+    right_bases: NotRequired[npt.NDArray[np.intp]]
+    width_heights: NotRequired[npt.NDArray[np.float64]]
+    left_ips: NotRequired[npt.NDArray[np.float64]]
+    right_ips: NotRequired[npt.NDArray[np.float64]]
+    plateau_sizes: NotRequired[npt.NDArray[np.intp]]
+    left_edges: NotRequired[npt.NDArray[np.intp]]
+    right_edges: NotRequired[npt.NDArray[np.intp]]
+
+def argrelmin(
+    data: npt.NDArray[np.generic], axis: int = 0, order: int = 1, mode: _Mode = "clip"
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def argrelmax(
+    data: npt.NDArray[np.generic], axis: int = 0, order: int = 1, mode: _Mode = "clip"
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def argrelextrema(
+    data: npt.NDArray[np.generic],
+    comparator: Callable[[npt.NDArray[np.generic], npt.NDArray[np.generic]], npt.NDArray[np.bool_]],
+    axis: int = 0,
+    order: int = 1,
+    mode: _Mode = "clip",
+) -> tuple[npt.NDArray[np.intp], ...]: ...
+def peak_prominences(x: npt.ArrayLike, peaks: npt.ArrayLike, wlen: int | None = None) -> _ProminencesResult: ...
 def peak_widths(
-    x: Untyped,
-    peaks: Untyped,
+    x: npt.ArrayLike,
+    peaks: npt.ArrayLike,
     rel_height: float = 0.5,
-    prominence_data: Untyped | None = None,
-    wlen: Untyped | None = None,
-) -> Untyped: ...
+    prominence_data: _ProminencesResult | None = None,
+    wlen: int | None = None,
+) -> _WidthsResult: ...
 def find_peaks(
-    x: Untyped,
-    height: Untyped | None = None,
-    threshold: Untyped | None = None,
-    distance: Untyped | None = None,
-    prominence: Untyped | None = None,
-    width: Untyped | None = None,
-    wlen: Untyped | None = None,
+    x: npt.ArrayLike,
+    height: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    threshold: float | npt.NDArray[np.float64] | tuple[float | None, float | None] | None = None,
+    distance: np.float64 | None = None,
```

Comment:
I'm guessing that (at least) `float` is also allowed here at runtime

Reply:
This is the same as `rel_height` where it is passed straight into a cython function which I didn't know before did automatic type conversion. This means using `AnyReal` should be fine I think.